cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -397,46 +398,48 @@ private Metadata getSnapshotBeforeTime(Id.NamespacedId targetId, long timeMillis
    */
   public List<MetadataEntry> search(final String namespaceId, final String searchQuery,
                                     final MetadataSearchTargetType type) {
-    Set<MetadataEntry> results = new HashSet<>();
+    List<MetadataEntry> results = new ArrayList<>();
```

Comment:
actually, let us not over-optimize. We will not do scoring until we start using an actual search engine. Here it would be sufficient to index each word only once even if it appears more than once.